feature_engine/variable_manipulation.py

```diff
@@ -52,10 +111,26 @@ def _find_categorical_variables(X, variables=None):
         if len(X[variables].select_dtypes(exclude='O').columns) != 0:
             raise TypeError("Some of the variables are not categorical. Please cast them as object "
```

Comment:
same here: should we replace this list comparison by a genexp?

Reply:
The default logic for ```variables``` is that the user should provide the list of strings. If the user provides only one variable as a string the function raises ```AttributeError``` due to a Pandas logic. Do we need to restrict the function to take the only a list of strings and not a string?
@solegalli

Reply:
I already replaced list comprehension by genexp